src/cycax/cycad/engine.py

```diff
@@ -1,13 +1,217 @@
+        )
+        return res
+
+    def decode_nut(self, lookup: dict)->str:
+        """
+        This method will return the string that will have the scad for a nut cut out.
+        
+        Args:
+            lookup : This will be a dictionary containing the necessary information about the nut.
+            
+        """
+        res = (
+            "rotate(["
+            + str(0)
+            + ", "
+            + str(0)
+            + ", "
+            + str(0)
+            + "])"
+            + "cylinder(r="
+            + str(lookup["nut_type"])
+            + ", h="
+            + str(lookup["depth"])
+            + ", $fn=6);"
+        )
+        return res
+
+    def decode_cut(self)->str:
+        """
+        This method returns a simple scad string neceseray to cut.
+        """
+        return "difference(){"
+
+    def translate(self, lookup: dict)->str:
+        """
+        This will move the object around and return the scad necessary.
+
+        Args:
+            lookup : This will be a dictionary containing the necessary information about the hole.
+        """
+        res = (
+            "translate(["
+            + str(lookup["x"])
+            + ", "
+            + str(lookup["y"])
+            + ", "
+            + str(lookup["z"])
+            + "])"
+        )
+        return res
+
+    def rotate(self, side: str)->str:
+        """
+        This will rotate the object and return the scad necessary.
+        
+        ???Would it make sense to also have a dictionary here similar to location swap methods???
+        
+        Args:
+            side : this is the side as retrieved form the dictionary. 
+        """
+        if side == TOP:
+            return "rotate([" + str(0) + ", " + str(180) + ", " + str(0) + "])"
+        elif side == BOTTOM:
+            return "rotate([" + str(0) + ", " + str(0) + ", " + str(0) + "])"
+        elif side == LEFT:
+            return "rotate([" + str(0) + ", " + str(90) + ", " + str(0) + "])"
+        elif side == RIGHT:
+            return "rotate([" + str(0) + ", " + str(270) + ", " + str(0) + "])"
+        elif side == FRONT:
+            return "rotate([" + str(270) + ", " + str(0) + ", " + str(0) + "])"
+        elif side == BACK:
+            return "rotate([" + str(90) + ", " + str(0) + ", " + str(0) + "])"
+        else:
+            msg = f"Side: {side} is not one of TOP, BOTTOM, LEFT, RIGHT, FRONT, BACK."
+            raise ValueError(msg)
+
+
+    def decode(self, data_file: str):
+        """
+        This is the main working class for decoding the scad. It is necessary for it to be refactored.
+        
+        !!!For this method to work properly it will be necessary to add a JSON, STL and SCAD file into the working repository.!!!
+        
+        Args: 
+            data_file : name of the file that is to be decoded into a scad.
+        """
+        out_name = "./SCAD/" + data_file + ".scad"
+        SCAD = open(out_name, "w")
```

Comment:
Consider using a context manager for this. Thus one any error you know the file is closed.